### PR TITLE
RavenDB-6978

### PR DESCRIPTION
**Pull request by mobilunity-user**

### CODE REVIEW

File: src/Database/Commands/GetClusterTopologyCommand.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import {GetTopologyCommand} from "./GetTopologyCommand";
+import {RequestMethods} from "../../Http/Request/RequestMethod";
+import {StatusCodes} from "../../Http/Response/StatusCode";
+import {StringUtil} from "../../Utility/StringUtil";
+
+export class GetClusterTopologyCommand extends GetTopologyCommand {
+  public createRequest(serverNode: ServerNode): void {
+    super.createRequest(serverNode);
+    this.removeParams('name');
+    this.endPoint = StringUtil.format('{url}/admin/cluster/topology', serverNode);
```
Comment (ayende) [inline src/Database/Commands/GetClusterTopologyCommand.ts:16]: This was just changed to `/cluster/topology`

File: src/Http/Request/RequestExecutor.ts [hunk 2]
```diff
@@ -7,79 +8,152 @@ import {IRavenResponse} from "../../Database/RavenCommandResponse";
...
     super();
 
+    const urls = options.firstTopologyUpdateUrls || [];
+
     this.headers = {
       "Accept": "application/json",
-      "Has-Api-key": TypeUtil.isNone(apiKey) ? 'false' : 'true',
-      "Raven-Client-Version": "4.0.0.0",
+      "Has-Api-key": 'false',
+      "Raven-Client-Version": "4.0.1.2",
```
Comment (gregolsky) [inline src/Http/Request/RequestExecutor.ts:71]: @a-serdukov-mobilunity Where did that number come from?

File: src/Http/Request/RequestExecutor.ts [hunk 3]
```diff
@@ -189,109 +237,137 @@ export class RequestExecutor extends Observable {
...
-          if (node.isFailed) {
-            setTimeout(this.updateFailingNodeStatus(node), 5 * 1000);
+    if (!this.isFirstTopologyUpdateTriesExpired()) {
+      this._fistTopologyUpdatesTries++;
+      this._firstTopologyUpdate = update(urls.pop());
+    }    
+  }
+
+  protected isFirstTopologyUpdateTriesExpired() {
+    return this._fistTopologyUpdatesTries >= this._maxFirstTopologyUpdatesTries;
```
Comment (gregolsky) [inline src/Http/Request/RequestExecutor.ts:293]: typo - `_fist..`. -> `_first...`

**Comment by gregolsky** [comment]
Also I get an error running tests:
```
 Index commands test
    Index actions
      √ should put index with success
      √ should get index with success (42ms)
      1) "before each" hook for "should get index with fail"


  75 passing (23s)
  1 failing

  1)  "before each" hook for "should get index with fail":
     ConcurrencyException: Raven.Client.Exceptions.ConcurrencyException: Database 'NorthWindTest' already exists!
   at Raven.Server.Web.System.AdminDatabasesHandler.<Put>d__4.MoveNext() in C:\work\ravendb4\src\Raven.Server\Web\System\AdminDatabasesHandler.cs:line 204
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Raven.Server.Routing.RequestRouter.<HandlePath>d__5.MoveNext() in C:\work\ravendb4\src\Raven.Server\Routing\RequestRouter.cs:line 118
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Raven.Server.RavenServerStartup.<RequestHandler>d__12.MoveNext() in C:\work\ravendb4\src\Raven.Server\RavenServerStartup.cs:line 158
      at Function.throw (.build\src\Utility\ExceptionThrower.js:19:15)
      at Function.throwFrom (.build\src\Utility\ExceptionThrower.js:32:31)
      at Function.throwFrom (.build\src\Utility\ExceptionThrower.js:25:22)
      at CreateDatabaseCommand.setResponse (.build\src\Database\RavenCommand.js:51:45)
      at CreateDatabaseCommand.setResponse (.build\src\Database\Commands\CreateDatabaseCommand.js:23:30)
      at RequestPromise.finally.catch.then.exception (.build\src\Http\Request\RequestExecutor.js:158:43)
      at tryCatcher (node_modules\bluebird\js\release\util.js:16:23)
      at Promise._settlePromiseFromHandler (node_modules\bluebird\js\release\promise.js:512:31)
      at Promise._settlePromise (node_modules\bluebird\js\release\promise.js:569:18)
      at Promise._settlePromise0 (node_modules\bluebird\js\release\promise.js:614:10)
      at Promise._settlePromises (node_modules\bluebird\js\release\promise.js:693:18)
      at Async._drainQueue (node_modules\bluebird\js\release\async.js:133:16)
      at Async._drainQueues (node_modules\bluebird\js\release\async.js:143:10)
      at Immediate.Async.drainQueues (node_modules\bluebird\js\release\async.js:17:14)
```

File: src/Http/Request/RequestExecutor.ts (hunk 2 above)
Comment (mobilunity-user) [reply src/Http/Request/RequestExecutor.ts:71]: from Python client

**Comment by mobilunity-user** [comment]
@gregolsky It's happens just sometimes, generally all tests passed.
Probably its some bug of current  Raven server? Or due to the some changes we need do something for example wait for delete database operation is complete? But there are no OperationID in response

File: src/Http/Request/RequestExecutor.ts (hunk 2 above)
Comment (gregolsky) [reply src/Http/Request/RequestExecutor.ts:71]: Let's keep it in sync with what's in *package.json* with this. Python client has different versioning pattern.

**Comment by gregolsky** [comment]
@a-serdukov-mobilunity OK, I'll create an issue for that. I believe this endpoint should just wait until db is deleted (https://github.com/ravendb/ravendb/blob/v4.0/src/Raven.Server/Web/System/AdminDatabasesHandler.cs#L613).

Meanwhile can you try workarounding it adding a `setTimeout()`/delay of some kind? Or by randomizing database name for tests?

File: src/Http/Request/RequestExecutor.ts (hunk 2 above)
Comment (ayende) [reply src/Http/Request/RequestExecutor.ts:71]: `Has-Api-Key` goes away

**Pull request merged**